prow/plank/plank.go

```diff
@@ -17,115 +17,40 @@ limitations under the License.
 package plank
 
 import (
-	"fmt"
 	"time"
 
+	"github.com/satori/go.uuid"
+
+	"k8s.io/test-infra/prow/config"
 	"k8s.io/test-infra/prow/kube"
-	"k8s.io/test-infra/prow/line"
 )
 
-type kubeClient interface {
-	ListJobs(map[string]string) ([]kube.Job, error)
-	ListProwJobs(map[string]string) ([]kube.ProwJob, error)
-	ReplaceProwJob(string, kube.ProwJob) (kube.ProwJob, error)
-	CreateJob(kube.Job) (kube.Job, error)
-}
-
-type Controller struct {
-	kc kubeClient
-}
-
-func NewController(kc *kube.Client) *Controller {
-	return &Controller{
-		kc: kc,
+func NewProwJob(spec kube.ProwJobSpec) kube.ProwJob {
+	return kube.ProwJob{
+		Metadata: kube.ObjectMeta{
+			Name: uuid.NewV1().String(),
+		},
+		Spec: spec,
+		Status: kube.ProwJobStatus{
+			StartTime: time.Now(),
+			State:     kube.TriggeredState,
```

Comment:
Seems `TriggeredState` is not really used. We can use `TriggeredState` to signal if the pod has not started yet, rather than rely on an empty pod name.